src/retrieve-validation/browser.ts

```diff
@@ -2,10 +2,36 @@
 import { W3CCSSValidatorResponse } from '.';
 
 // Utility function for retrieving response from W3C CSS Validator in a browser environment
-const retrieveInBrowser = async (url: string): Promise<W3CCSSValidatorResponse['cssvalidation']> => {
-	const res = await fetch(url);
+const retrieveInBrowser = async (url: string, timeout: number): Promise<W3CCSSValidatorResponse['cssvalidation']> => {
+	// Initialize controller who's signal will abort the fetch
+	const controller = new AbortController();
+
+	// Start timeout
+	setTimeout(() => {
+		controller.abort();
+	}, timeout);
+
+	// Attempt to fetch CSS validation, catching the abort error to handle specially
+	let res: Response | null = null;
+
+	try {
+		res = await fetch(url, { signal: controller.signal });
+	} catch (err) {
+		if ((err as Error).name === 'AbortError') {
+			throw new Error(`The request took longer than ${timeout}ms`);
+		}
+
+		throw new Error(err);
```

Comment:
We can probably just throw this directly, right? I'm not sure if wrapping it in `Error()` would impact the trace or call stack.

```suggestion
		throw err;
```

Reply:
This will probably work, if for some reason "err" is not an instance of the Error class, there won't be a stack trace.